server/src/__tests__/analyzer.test.ts

```diff
@@ -0,0 +1,106 @@
+import { generateParser } from '../tree-sitter/parser'
```

Comment:
You should add a commit that runs this in CI! does `npm run test` run this?

Reply:
yes, I can add an action in CI once this is good to merge. Running `npm run test` will execute all tests in `__test__`